check_isa.cpp

```diff
@@ -19,7 +19,7 @@
 #define HOST_IS_APPLE
 #endif
 
-#if !defined(__arm__) && !defined(__aarch64__) && !defined(_M_ARM64)
+#if defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || defined(_M_AMD64)
```

Comment:
`_M_AMD64` and `_M_X64` are equivalent according to this [blog](https://learn.microsoft.com/en-us/archive/blogs/reiley/macro-revisited). I would prefer `_M_X64`, it's just more clear naming.

Reply:
CI looks good